### PR TITLE
treewide: fix typos and false positives found by codespell

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

This fixes some typos and adds one correctly used abbreviation to the ignore list.

### Testing procedure

If bors merges this without static tests failing, it did what it intended to do.

### Issues/PRs references

Fixes issues that prevent https://github.com/RIOT-OS/RIOT/pull/19440 from merging

### CODE REVIEW

**Comment by maribu** [comment]
Maybe it makes sense to squash this into three commits:

1. `treewide: fix typos in comments`
2. `sys/ztimer: fix typo in debug output`
3. `dist/tools/codespell: ignore false positives`

File: dist/tools/codespell/ignored_words.txt [hunk 1]
```diff
@@ -17,6 +17,9 @@ dout
 # ALS (Ambient Light Sensing) => ALSO
 als
 
+# CAF (Comp. A Enable Output Filter (MSP430) or Copy All Frames (SAM0) or C++ Actor Framework) => CALF
+CAF
```
Comment (maribu) [inline dist/tools/codespell/ignored_words.txt:21]:
```suggestion
caf
```

Uppercase doesn't work?

Comment (maribu) [reply dist/tools/codespell/ignored_words.txt]: Indeed, now static tests are passing

**Review by aabadie — APPROVED** [review]
ACK

**Comment by aabadie** [comment]
bors merge

**Pull request merged**